riva/clients/asr/client_call.cc

```diff
@@ -13,45 +13,63 @@ ClientCall::ClientCall(uint32_t corr_id, bool word_time_offsets)
   recv_final_flags.reserve(1000);
 }
 
+ClientCall::~ClientCall() {
+  if (pipeline_states_logs_)
+    pipeline_states_logs_.close();
+}
+
 void
 ClientCall::AppendResult(const nr_asr::StreamingRecognitionResult& result)
 {
-  bool is_final = result.is_final();
-  if (latest_result_.final_transcripts.size() < 1) {
-    latest_result_.final_transcripts.resize(1);
-    latest_result_.final_transcripts[0] = "";
-  }
-
-  if (is_final) {
-    int num_alternatives = result.alternatives_size();
-    latest_result_.final_transcripts.resize(num_alternatives);
-    latest_result_.final_scores.resize(num_alternatives);
-    latest_result_.final_time_stamps.resize(num_alternatives);
-    for (int a = 0; a < num_alternatives; ++a) {
-      // Append to transcript
-      latest_result_.final_transcripts[a] += result.alternatives(a).transcript();
-      latest_result_.final_scores[a] += result.alternatives(a).confidence();
+  if (result.has_pipeline_states()) {
+    auto pipeline_states = result.pipeline_states();
+    int prob_states_count = pipeline_states.vad_probabilities_size();
```

Comment:
Technically size() returns size_t datatype. If the vad_probabilities_size() hits int limits, it would roll over and return negative values. It's better to use size_t instead of int.